shortest_path/dijkstras.py

```diff
@@ -47,7 +44,7 @@ def dijkstras_heap(G, start='A'):
 
 def dijkstras(G, start='A'):
     shortest_paths = {}
-    unvisited = list(G.keys())
+    unvisited = dict.fromkeys(G.keys())
```

Comment:
The code used to be a list, which would then call remove on `min_node` in a loop. There's no guarantee that `min_node` is at the end of the list, so `remove` has to do a seek to find the `min_node` in the list and remove it, and both operations are O(n). 

If it's changed to a dictionary, finding the `min_node` is O(1) and removing it is also O(1).